Project 5 For John Beverly

```diff
@@ -51,3 +51,441 @@ You may find the following resources useful:
+] .
+
+
+
+
+8.	Constraint for disjoint of symmetry or asymmetry 
+a.	Our constraint checks for when a graph is either symmetric or asymmetric 
+
+b.	An example of when this constraint could be used to check a dataset of products represented as an RDF resource with various properties. This constraint will ensure a specific property (or multiple properties) is/are always present when a product is listed. To take this further, let’s say that our product type must have a review and a rating as the property, when the graph is checking the dataset it will check for symmetry of review existing and rating existing while also checking for an asymmetry such that there is a review but no rating or a rating and no review. One would change ex:symmetricalProperty to ex:hasReviewShape and then following this basic formatting!
+
+c.	[(a -> b) ^ (b -> a)] v [(a -> b) ^ ~(b -> a)]
+
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+
+@prefix ex: <http://example.com/> .
+
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+
+ex:AsymReflexShape
+
+a sh:NodeShape ;
+
+sh:targetClass ex:relation ;
+
+sh:property [    
+
+sh:path ex:hasSource ;    
+
+sh:class ex:Node 
+
+] ;
+
+sh:property [    
+
+
+sh:path ex:hasTarget ;    
+
+sh:class ex:Node  
+
+] ;
+
+sh:property [    
+
+sh:path ex:hasValue ;    
+
+sh:datatype xsd:boolean ;    
+
+sh:in ( true )  ] ;
+
+sh:property [    
+
+sh:path ex:hasType ;    
+
+sh:in ( "asymmetric" )  
+
+] .
+
+
+9.	Constraint for PropertyIn for Asymmetrical or Irreflexive 
+a.	Our constraint checks for when a graph has a propertyIn that is both asymmetric and irreflexive. 
+
+b.	An example of a time when you could use this constraint would be when checking a flow chart type graph (I think they are called directed acyclic graphs). Such that in a flow chart there could be many outgoing relations with only one incoming relation. This means the ex:asymmetricalPropertySHape constraint could ensure that each arrow of the graph in directed in a consistent way. While is ex:IrreeflexivePropertyShape constraint on the other hand, could be used to ensure that there are no cycles in the graph by prohibiting self-reference. By combing these constraints with sh:or property type you can check the validity of a flow chart. 
+
+c.	[(a -> b) ^ ~(b -> a)] v [~(a -> a)]
+
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix ex: <http://example.com/> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+ex:asymmetricalPropertyShape
+
+a sh:PropertyShape ;
+
+sh:path ex:asymmetricalProperty ;
+
+sh:propertyConstraint [
+
+sh:asymmetrical true ;
+
+] .
+
+ex:irreflexivePropertyShape
+
+a sh:PropertyShape ;
+
+sh:path ex:irreflexiveProperty ;
+
+sh:propertyConstraint [
+
+sh:irreflexive true ;
+
+] .
+
+ex:MyClassShape
+
+a sh:NodeShape ;
+
+sh:property [
+
+sh:or ( ex:asymmetricalPropertyShape ex:irreflexivePropertyShape ) ;
+
```

Comment:
I'm wondering whether we can use "sh:or" like this. sh:or can be used to include several nodes (including blank nodes like "[ sh:path ex:givenName ; sh:minCount 1 ; ]" ), but can it be used to directly include a node shape?

Reply:
Yes, you could use sh:or to include several node shapes. The sh:or property takes a list of node shapes as its values and any data that conforms to at least one of the node shapes in the list would be valid. Let's say we have three node shapes ex:Shape1, ex:Shape2, and ex:Shape3 you can use sh:or to combine them like this 
ex:CombinedShape
  a sh:NodeShape ;
  sh:or ( ex:Shape1 ex:Shape2 ex:Shape3 ) .

ex:CombinedShape is a new node shape that includes the three node shapes ex:Shape1, ex:Shape2, and ex:Shape3 as alternatives. Any data that conforms to at least one of these node shapes will be considered valid. The node shapes included in the sh:or list can be simple or complex, and can include blank nodes as well. 

I hope this helps :)

Reply:
Goood to know about this, thanks!